mm/mm_gran/mm_granreserve.c

```diff
@@ -54,13 +54,15 @@
  *   size   - The size of the region to be reserved
  *
  * Returned Value:
- *   None
+ *   On success, a non-NULL pointer to the allocated memory is returned;
+ *   NULL is returned on failure.
  *
  ****************************************************************************/
 
-void gran_reserve(GRAN_HANDLE handle, uintptr_t start, size_t size)
+FAR void *gran_reserve(GRAN_HANDLE handle, uintptr_t start, size_t size)
 {
   FAR struct gran_s *priv = (FAR struct gran_s *)handle;
+  FAR void *ret = (FAR void *)start;
```

Comment:
```suggestion
  FAR void *ret = NULL;
```

Reply:
Philosophical argument: what should happen when this is called with size = 0 (like is tested on line 69)?

In my opinion it can just as well succeed. But I can change this to NULL as well

Reply:
Done